UniWatch/Views/Lecture/Index.cshtml

```diff
@@ -7,13 +7,67 @@
 <h2>@Model.ClassName</h2>
 <hr />
 
-@{
+@if (Request.IsAuthenticated && User.IsInRole("Student"))
+{
```

Comment:
Similar to the other view, there is a lot of code duplication between the student and teacher view. Can this be minimized?